nucypher/utilities/porter/control/specifications/fields/treasuremapid.py

```diff
@@ -19,11 +19,14 @@
 
 from nucypher.control.specifications.exceptions import InvalidInputData
 from nucypher.control.specifications.fields.base import BaseField
+from nucypher.crypto.constants import HRAC_LENGTH
 from nucypher.policy.collections import TreasureMap
 
 
 class TreasureMapID(BaseField, fields.String):
 
     def _validate(self, value):
-        if len(bytes.fromhex(value)) != TreasureMap.ID_LENGTH:
+        treasure_map_id = bytes.fromhex(value)
+        # FIXME federated has map id length 32 bytes but decentralized has length 16 bytes ... huh? - #2725
```

Comment:
Needs to be revised with special consideration towards revocation.  16 bytes is the length of the field in `PolicyManager`, so let's stick to that unless we plan to modify the contract.  HRAC / Policy ID length must be the first 16 bytes, truncated from the keccack digest regardless of operating mode.